test/fixtures/rules/std_rule_cases/L052.yml

```
@@ -464,3 +464,12 @@ test_fail_newline_create_table_inline_comment:
     rules:
       L052:
         multiline_newline: true
+
+test_fail_whitespace_after_snowflake_set:
+  fail_str: |
+    SET foo = (SELECT foo FROM foo.foo) ;
+  fix_str: |
+    SET foo = (SELECT foo FROM foo.foo);
+  configs:
+    core:
+      dialect: snowflake
```

The simpler statement `SELECT 1  ;` worked before and autofixed.

So that clearly means it went down a different code path (cause it's actually `SELECT<indent>1<dedent> ;` and those `indents/dendents` seemed to be enough to allow it to auto fix with the old code.

Therefore, as it's going down a different code path (or at least used to!), I think we should add that as a test case too.

```suggestion
test_fail_whitespace_after_simple_select:
  fail_str: |
    SELECT 1 ;
  fix_str: |
    SELECT 1;
 
test_fail_whitespace_after_snowflake_set:
  fail_str: |
    SET foo = (SELECT foo FROM foo.foo) ;
  fix_str: |
    SET foo = (SELECT foo FROM foo.foo);
  configs:
    core:
      dialect: snowflake
```